habitat-lab/habitat/datasets/rearrange/samplers/object_sampler.py

```diff
@@ -376,7 +394,12 @@ def single_sample(
 
         return new_object, target_receptacle
 
-    def set_num_samples(self):
+    def set_num_samples(self, num_objects: Tuple[int, int] = None):
+        if num_objects is not None:
+            self.num_objects = num_objects
+            assert self.num_objects[1] >= self.num_objects[0]
```

Comment:
This assert should have a descriptive message.

Reply:
I need to rework these changes (dynamic object range), I've implemented it in a hacky way right now.

Reply:
Add TODOs

Reply:
I've fixed this implementation.